Jenkinsfile.promote

```diff
@@ -77,34 +81,126 @@ pipeline {
             when {
                 expression { return isRelease() }
             }
             steps{
                 script {
-                    dir('optaplanner') {
-                        checkoutRepo('optaplanner')
-                        mergeAndPush('optaplanner', getDeployPrLink('optaplanner'))
+                    dir(optaplannerRepository) {
+                        checkoutRepo(optaplannerRepository)
+                        mergeAndPush(getDeployPrLink(optaplannerRepository))
+                        tagLatest()
+                    }
+                }
+            }
+        }
+
+        stage('Merge Optaweb Vehicle Routing deploy PR and tag') {
+            when {
+                expression { return isRelease() }
+            }
+            steps{
+                script {
+                    dir(vehicleRoutingRepository) {
+                        checkoutRepo(vehicleRoutingRepository)
+                        mergeAndPush(getDeployPrLink(vehicleRoutingRepository))
+                        tagLatest()
+                    }
+                }
+            }
+        }
+
+        stage('Merge Optaweb Employee Rostering deploy PR and tag') {
+            when {
+                expression { return isRelease() }
+            }
+            steps{
+                script {
+                    dir(employeeRosteringRepository) {
+                        checkoutRepo(employeeRosteringRepository)
+                        mergeAndPush(getDeployPrLink(employeeRosteringRepository))
                         tagLatest()
                     }
                 }
             }
         }
 
-        stage('Upload documentation and distribution') {
+        stage('Upload OptaPlanner documentation and distribution') {
             when {
                 expression { return isRelease() }
             }
             steps {
                 script {
-                    mavenCleanInstall('optaplanner', true, [], '-Dfull')
-                    dir('optaplanner') {
+                    dir(optaplannerRepository) {
+                        mavenCleanInstall(true, [], '-Dfull')
                         withCredentials(bindings: [sshUserPrivateKey(credentialsId: 'optaplanner-filemgmt',
                                              keyFileVariable: 'SSH_KEY_JBOSS_FILEMGMT')]) {
                             // For testing, simulate connection via SSH:
-                            // sh "ssh -i $SSH_KEY_JBOSS_FILEMGMT -oKexAlgorithms=+diffie-hellman-group1-sha1 optaplanner@filemgmt.jboss.org"
-                            sh "./build/release/upload_distribution.sh ${getProjectVersion()} $SSH_KEY_JBOSS_FILEMGMT"
+                            sh "ssh -i $SSH_KEY_JBOSS_FILEMGMT -oKexAlgorithms=+diffie-hellman-group1-sha1 optaplanner@filemgmt.jboss.org"
+                            // TODO: return back
+                            //sh "./build/release/upload_distribution.sh ${getProjectVersion()} $SSH_KEY_JBOSS_FILEMGMT"
```

Comment:
I say this (ssh + script call) could be factorized as you are executing 3 times the same or ?
even better as you are also using the creds part